cmd/cctool/unpack.go

```diff
@@ -0,0 +1,137 @@
+package main
+
+import (
+	"bufio"
+	"bytes"
+	"context"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strconv"
+	"time"
+
+	"github.com/quay/claircore/internal/indexer/fetcher"
+)
+
+type unpackConfig struct {
+	timeout time.Duration
+}
+
+// Unpack will decompress and untar each layer of the provided image ref
+func Unpack(cmd context.Context, cfg *commonConfig, args []string) error {
+	cmdcfg := unpackConfig{}
+	fs := flag.NewFlagSet("cctool unpack", flag.ExitOnError)
+	fs.DurationVar(&cmdcfg.timeout, "timeout", 5*time.Minute, "timeout for successful http responses")
+	fs.Usage = func() {
+		out := fs.Output()
+		fmt.Fprintf(out, "Usage:\n")
+		fmt.Fprintf(out, "\tcctool unpack <image-ref>\n")
+		fmt.Fprintf(out, "Arguments:\n")
+		fmt.Fprintf(out, "\timage-ref: a reference to an image and it's repository\n\n")
+	}
+	fs.Parse(args)
+
+	// get the image reference from argument.
+	if len(fs.Args()) <= 0 || len(fs.Args()) >= 2 {
+		fs.Usage()
+		return nil
+	}
+
+	image := fs.Args()[0]
+	ctx, done := context.WithTimeout(cmd, cmdcfg.timeout)
+	defer done()
+
+	// inspect image reference and get manifest
+	m, err := Inspect(ctx, image)
+	if err != nil {
+		return err
+	}
+
+	// signal to main cli we need to wait on cleanup
+	// we are about to write to the file system
+	cfg.cleanup.Add(1)
+
+	// fetch layers
+	ctx, done = context.WithTimeout(cmd, cmdcfg.timeout)
+	defer done()
+
+	log.Printf("fetching layers")
+	f := fetcher.New(nil, "")
+	err = f.Fetch(ctx, m.Layers)
+	if err != nil {
+		return err
+	}
+	log.Printf("successfully fetched layers")
+
+	// create a tmp dir we will unpack layers to
+	td, err := ioutil.TempDir("", "cctool-unpack-")
+
+	log.Printf("exacting layers into tmp dir: %v.", td)
+	for i, layer := range m.Layers {
+		dirName := filepath.Base(layer.Local())
+		dir := filepath.Join(td, strconv.Itoa(i)+"-"+dirName)
```

Comment:
I think it'd be easier to use the layer digest instead of grabbing the local file name.